### ISSUE TITLE
Ability to get and answer questions a VM has pending

### ISSUE THREAD

**Issue by dhildebrandt** [chef-partners/knife-vsphere#308]
This is a feature request; below in the issue template I have filled out what I was trying to do when I realized I wanted the feature.

The feature request is for the ability to deal with questions the VM poses.  I would like to see the ability to find out whether there are questions, what the available responses are, any text that the hypervisor has provided so that people or automation could deal with it, and then the ability to respond to the question using knife vsphere vm.

I see something like the following as desirable

`knife vsphere vm question get VMNAME` - return the question and available responses in some machine parsable format

`knife vsphere vm question respond VMNAME <response>` - supplies the response to the VM

### What are you trying to do and what's not working?
TL;DR: reconstruct a cluster on a nightly basis and have the ability to deal with VM questions.

Long Answer: My nightly build powers off my test cluster and deletes its vms before rebuilding it all.  My hypervisor storage filled up and one of the knife vsphere commands (either poweroff or delete) got the exception below.  Eventually my retry code gave up, and the cluster deployment failed.

### Please paste the command and output (run with `-VV`). Don't forget to X out any passwords or sensitive information

The command was:
```
kv vm state $target_host -f $folder -s off
```
where
`folder`=the vcenter folder the machine is in
`target_host`=VMNAME

note that "kv" is a shell function, the salient part of which is
```
knife vsphere "$@" -u "$VCENTER_USER" -p "$VCENTER_PASS" --vshost "$VCENTER_HOST" --vsdc "$VCENTER_DATACENTER" --vsinsecure;
```
If the condition happens again , I will remember to re-run with -VV and post the results here.

Here is my log excerpt.  I have omitted all the retries except the last for brevity.  I believe that the "state" command operated correctly here and that is why I'm requesting the abililty to query and respond to questions.
```
21:22:17 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: Powering sc-cap-ci1-app2 off
21:22:32 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: ERROR: RbVmomi::Fault: QuestionPending: The operation cannot be allowed at the current time because the virtual machine has a question pending:
21:22:32 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: 'msg.hbacommon.outofspace:There is no more space for virtual disk sc-cap-ci1-app2.vmdk. You might be able to continue this session by freeing disk space on the relevant volume, and clicking Retry. Click Cancel to terminate this session.
21:22:32 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: '.
21:22:33 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: Retry failed [5]: kv vm state sc-cap-ci1-app2 -f Socialcast/_CI/prototype -s off
21:22:33 08_teardown.sh cluster-teardown-expansion-sc-cap-ci1-app2: err: could not poweroff vm sc-cap-ci1-app2
```

### Other details, such as gem version and if you're working with a Linux or Windows VM
Using knife-vsphere 1.0.1
on Ubuntu 14.04.5 LTS
talking to ESXi 5.1.0
via Vsphere 5.1.0

chef itself is not in the mix: this is just plain knife vsphere

**Comment by swalberg** [comment]
Let me look around to see what's in the API. Thanks for the detailed description!

**Comment by dhildebrandt** [comment]
Really appreciate your work on this project.
